src/lib/log/audit/audit.go

```diff
@@ -0,0 +1,148 @@
+//  Copyright Project Harbor Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package audit
+
+import (
+	"context"
+	"fmt"
+	"log/syslog"
+	"math/rand"
+	"strings"
+	"time"
+
+	"github.com/goharbor/harbor/src/lib/cache"
+	"github.com/goharbor/harbor/src/lib/config"
+	"github.com/goharbor/harbor/src/lib/log"
+)
+
+const (
+	endpointHealthCheckInterval = 1 * time.Minute
+	cacheInterval               = 2 * time.Minute
+)
+
+var LogMgr = &LoggerManager{}
+
+// LoggerManager manage the operations related to the audit log
+type LoggerManager struct {
+	currentEndpoint string
+	cache           cache.Cache
+	auditLogger     *log.Logger
+}
+
+// Init redirect the audit log to the forward endpoint
+func (a *LoggerManager) Init(ctx context.Context, logEndpoint string) {
+	a.cache = cache.Default()
+	a.currentEndpoint = logEndpoint
+	al, err := syslog.Dial("tcp", logEndpoint,
+		syslog.LOG_INFO, "audit")
+	if err != nil {
+		log.Errorf("failed to create audit log, error %v", err)
+		a.SetEndpointStatus(ctx, logEndpoint, false)
+		return
+	}
+	a.auditLogger = log.New(al, log.NewTextFormatter(), log.InfoLevel, 3)
+	a.SetEndpointStatus(ctx, logEndpoint, true)
+}
+
+// DefaultLogger ...
+func (a *LoggerManager) DefaultLogger(ctx context.Context) *log.Logger {
+	endpoint := config.AuditLogForwardEndpoint(ctx)
```

Comment:
can we just use it and not update any values? If the server can't connect, let it fail.